sdk/src/trace/span.cc

```diff
@@ -63,7 +64,7 @@ void Span::End() noexcept
   {
     return;
   }
-  tracer_->recorder().Record(std::move(recordable_));
+  tracer_->GetProcessor().OnEnd(std::move(recordable_));
```

Comment:
If we want to ensure that OnEnd is called on the same span as OnStart, we'd need for the span to store a `std::shared_ptr<SpanProcessor>`. Otherwise, the span processor could be changed between start and end.

Reply:
Do you think we need to store both the processor and the [tracer](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/trace/span.h#L108)? Or do you think we could go just with storing the processor and omitting the tracer (and the tracer accessor)?

Reply:
Getting the tracer from a span is something that's in all the language apis

Reply:
I don't see it in the [Python](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-api/src/opentelemetry/trace/__init__.py#L173) or [Java](https://github.com/open-telemetry/opentelemetry-java/blob/master/api/src/main/java/io/opentelemetry/trace/Span.java#L34) APIs. But maybe I'm missing something here.

For now I added both shared pointers to the tracer and to the processor to the span.

Reply:
Interesting...

I see it on the go api
https://github.com/open-telemetry/opentelemetry-go/blob/master/api/trace/api.go#L88

Does the spec say anything about it?

Reply:
> Does the spec say anything about it?

I couldn't find anything in the spec about getting Tracer from a Span.
However from the implementation perspective, I think it makes sense for the Span to have a reference to the list of SpanProcessors, and I think it is fine to expose it from the SDK, but not API).

FYI - this is what Python did:
* Span has a reference to the processor (which could be a `MultiSpanProcessor`, which is basically a list) https://github.com/open-telemetry/opentelemetry-python/blob/8c360a182d61a8694904bbcc99ef9f60a28e342a/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L244
* https://github.com/open-telemetry/opentelemetry-python/blob/8c360a182d61a8694904bbcc99ef9f60a28e342a/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L698